src/coreclr/src/jit/importer.cpp

```diff
@@ -6738,9 +6738,16 @@ void Compiler::impPopArgsForUnmanagedCall(GenTree* call, CORINFO_SIG_INFO* sig)
         assert(thisPtr->TypeGet() == TYP_I_IMPL || thisPtr->TypeGet() == TYP_BYREF);
     }
 
-    for (GenTreeCall::Use& use : GenTreeCall::UseList(args))
+    for (GenTreeCall::Use& argUse : GenTreeCall::UseList(args))
     {
-        call->gtFlags |= use.GetNode()->gtFlags & GTF_GLOB_EFFECT;
+        // We should not be passing gc typed args to an unmanaged call.
+        GenTree* arg = argUse.GetNode();
+        if (varTypeIsGC(arg->TypeGet()))
+        {
+            assert(!"*** invalid IL: gc type passed to unmanaged call");
```

Comment:
Is it ok for this to be ignored in Release builds? Or should this be a BADCODE?

Reply:
That is a good question. I took the guidance from https://github.com/dotnet/runtime/issues/34279#issuecomment-613695776.

Reply:
Given how many places we had to fix in our stuff, I would be very worried about breaking potential of turning this into BADCODE and the long tail of issues from that. It may be better to look into making JIT resilient to this.